api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -455,58 +456,59 @@ protected boolean containsAlias(Iterator<CriteriaImpl.Subcriteria> subcriteriaIt
 		});
 	}
 	
-	protected void handleLocationReference(Criteria criteria, ReferenceParam locationReference) {
+	protected void handleLocationReference(Criteria criteria, ReferenceAndListParam locationReference) {
 		if (locationReference != null) {
 			criteria.createAlias("location", "l");
 			
-			if (locationReference.getChain() != null) {
-				switch (locationReference.getChain()) {
+			handleAndListParam(locationReference, token -> {
+				switch (token.getChain()) {
 					case Location.SP_ADDRESS_CITY:
-						propertyLike("l.cityVillage", locationReference.getValue()).ifPresent(criteria::add);
+						propertyLike("l.cityVillage", token.getValue()).ifPresent(criteria::add);
 						break;
 					case Location.SP_ADDRESS_STATE:
-						propertyLike("l.stateProvince", locationReference.getValue()).ifPresent(criteria::add);
+						propertyLike("l.stateProvince", token.getValue()).ifPresent(criteria::add);
 						break;
 					case Location.SP_ADDRESS_POSTALCODE:
-						propertyLike("l.postalCode", locationReference.getValue()).ifPresent(criteria::add);
+						propertyLike("l.postalCode", token.getValue()).ifPresent(criteria::add);
 						break;
 					case Location.SP_ADDRESS_COUNTRY:
-						propertyLike("l.country", locationReference.getValue()).ifPresent(criteria::add);
-						break;
+						propertyLike("l.country", token.getValue()).ifPresent(criteria::add);
 					case "":
-						criteria.add(eq("l.uuid", locationReference.getValue()));
+						criteria.add(eq("l.uuid", token.getValue()));
 						break;
 				}
-			}
+				
+				return Optional.empty();
```

Comment:
I'd prefer this method to be re-written along the lines I requested for `handleEncounterReference` above. We don't want to be passing around the `Criteria` object unless we absolutely need to.